### DISCUSSION TITLE
Album Art as base64 (mqtt)

### DISCUSSION THREAD

**Discussion by ThunderBirrddd** [mikebrady/shairport-sync#1934] [Q&A]
Hey,
is it possible to fetch the album art in base64 encoded format instead of img entity, using mqtt. 
I had tried it myself but didn't succeed.

**Comment by mikebrady** [comment]
Thanks for the question. I’m not familiar enough with MQTT to know what the standard is for encoding images ( is there one?), but the artwork is in Base64 format in the metadata pipe, if that’s of any use.

**Reply by ThunderBirrddd** [reply]
oh so like if i try to fetch the album art through mqtt it appears to be img is there any way i can get the base64 version? 

actually i have shairport sync working inside of home assistant

**Reply by mikebrady** [reply]
What I wonder is whether there is a standard format for images in Home Assistant. Maybe @aaronk6 might have a view?

**Reply by aaronk6** [reply]
In theory, it should be possible to achieve this with an MQTT camera entity (see https://www.home-assistant.io/integrations/camera.mqtt/). It can handle binary data received via MQTT, so no base64 encoding would be necessary. I’ll see if I can get that up and running.

**Reply by aaronk6** [reply]
It works.

<img src="https://github.com/user-attachments/assets/8cd110d3-d5c3-4f84-951c-cea0c1351873" width="300" />

However, it seems to always show the previous artwork, not the current one. This must be something in Home Assistant. The image I get when subscribing manually to the topic is the correct one:

```
mosquitto_sub -h homeassistant.example.com -t "shairport-sync/bad_dg/cover" -C 1 > cover.jpg
```

@ThunderBirrddd Can you test this on your end and report back what results you get?

**Reply by aaronk6** [reply]
Looks like it’s a known issue in Home Assistant: https://github.com/home-assistant/core/issues/75930 I have quickly skimmed through this thread and it seems the suggested solution is sending two snapshots. That’s unfortunate.

**Reply by aaronk6** [reply]
Actually, scrap that—there is also an MQTT Image entity, which I totally did not have on my radar.

```
mqtt:
  - image:
      image_topic: shairport-sync/bad_dg/cover
```

This works perfectly here and updates almost instantly.

@mikebrady I’ll add this to my setup and see if it’s also stable over time. If so, I’ll make a PR to add it to the MQTT auto discovery payload (if `publish_cover` is enabled).

**Reply by mikebrady** [reply]
Fantastic, thanks.
